internal/cmd/beta/image/create/create.go

```diff
@@ -0,0 +1,401 @@
+	Config              *imageConfig
+	MinDiskSize         *int64
+	MinRam              *int64
+	Protected           *bool
+	NoProgressIndicator *bool
+}
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Creates images",
+		Long:  "Creates images.",
+		Args:  args.NoArgs,
+		Example: examples.Build(
+			examples.NewExample(
+				`Create a named image 'my-new-image' from a raw disk image located in '/my/raw/image'`,
+				`$ stackit beta image create --name my-new-image --disk-format=raw --local-file-path=/my/raw/image`,
+			),
+			examples.NewExample(
+				`Create a named image 'my-new-image' from a qcow2 image read from '/my/qcow2/image' with labels describing its contents`,
+				`$ stackit beta image create --name my-new-image --disk-format=qcow2 --local-file-path=/my/qcow2/image--labels os=linux,distro=alpine,version=3.12`,
+			),
+		),
+		RunE: func(cmd *cobra.Command, _ []string) (err error) {
+			ctx := context.Background()
+			model, err := parseInput(p, cmd)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			// we open input file first to fail fast, if it is not readable
+			file, err := os.Open(model.LocalFilePath)
+			if err != nil {
+				return fmt.Errorf("create image: file %q is not readable: %w", model.LocalFilePath, err)
+			}
+			defer func() {
+				if inner := file.Close(); inner != nil {
+					err = fmt.Errorf("error closing input file: %w (%w)", inner, err)
+				}
+			}()
+
+			if !model.AssumeYes {
+				prompt := fmt.Sprintf("Are you sure you want to create the image %q?", model.Name)
+				err = p.PromptForConfirmation(prompt)
+				if err != nil {
+					return err
+				}
+			}
+
+			// Call API
+			request := buildRequest(ctx, model, apiClient)
+
+			result, err := request.Execute()
+			if err != nil {
+				return fmt.Errorf("create image: %w", err)
+			}
+			model.Id = result.Id
+			url, ok := result.GetUploadUrlOk()
+			if !ok {
+				return fmt.Errorf("create image: no upload URL has been provided")
+			}
+			if err := uploadAsync(ctx, p, model, file, *url); err != nil {
+				return err
+			}
+
+			if err := outputResult(p, model, result); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	configureFlags(cmd)
+	return cmd
+}
+
+func uploadAsync(ctx context.Context, p *print.Printer, model *inputModel, file *os.File, url string) error {
+	stat, err := file.Stat()
+	if err != nil {
+		return fmt.Errorf("upload file: %w", err)
+	}
+
+	var reader io.Reader
+	if model.NoProgressIndicator != nil && *model.NoProgressIndicator {
+		reader = file
+	} else {
+		var ch <-chan int
+		reader, ch = newProgressReader(file)
+		go func() {
+			ticker := time.NewTicker(2 * time.Second)
+			var uploaded int
+			done:
+			for {
+				select {
+				case <-ticker.C:
+					p.Info("uploaded %3.1f%%\n", 100.0/float64(stat.Size())*float64(uploaded))
```

Comment:
You can replace the `\n` with `\r`, then it doesn't print always a new line. Instead it would replace the previous output
```suggestion
					p.Info("uploaded %3.1f%%\r", 100.0/float64(stat.Size())*float64(uploaded))
```

Reply:
done